### PR TITLE
LPS-181943 | Test Automation | Master

### PR DESCRIPTION
**Pull request by victorhcunha**
[LPS-181943](https://issues.liferay.com/browse/LPS-181943) ObjectActions#CanAddCommerceProductGroupEntryAfterDeletingCommerceProductEntry
[LPS-181944](https://issues.liferay.com/browse/LPS-181944) ObjectActions#CanUpdateCommerceProductEntryAfterUpdatingCommerceProductEntry

Using reviews from #753

### CODE REVIEW

**Comment by victorhcunha** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 1]
```diff
@@ -106,6 +106,64 @@ definition {
...
+
+		task ("Given a minium site and Commerce Product object") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+
+			ApplicationsMenu.gotoSite(site = "Minium");
+
+			ApplicationsMenu.gotoPortlet(
+				category = "Product Management",
+				panel = "Commerce",
+				portlet = "Products");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:122]: You could use: `CommerceProducts.openProductsAdmin();`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 2]
```diff
@@ -106,6 +106,64 @@ definition {
...
+
+			CPCommerceCatalog.newProductsEntry(
+				productCatalog = "Minium",
+				productName = "Simple T-Shirt",
+				productType = "Simple");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:127]: Verify if is works with: `CommerceJSONProductsAPI._addCommerceProduct`

There are simple examples with it, we don't have to access and add the product via UI.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 3]
```diff
@@ -106,6 +106,64 @@ definition {
...
+				inputValue = "Value Test",
+				objectLabel = "Commerce Product Group",
+				thenAction = "Add an Object Entry",
+				whenAction = "On After Delete");
+		}
+
+		task ("When the user deletes a Commerce Product entry") {
+			CommerceNavigator.gotoPortlet(
+				category = "Product Management",
+				portlet = "Products");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:151]: Access via URL: `CommerceProducts.openProductsAdmin();`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 4]
```diff
@@ -3628,6 +3628,73 @@ definition {
 			actionLabel = "Update Action Label");
 	}
 
+	@description = "LPS-173537 - Verify updating a Commerce Product entry triggers an action to update the Commerce Product entry"
+	@priority = 5
+	test CanUpdateCommerceProductEntryAfterUpdatingCommerceProductEntry {
+		property portal.acceptance = "true";
+
+		task ("Given a minium site and Commerce Product") {
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:3636]: `task ("Given a minium site and Commerce Product object")`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 5]
```diff
@@ -3628,6 +3628,73 @@ definition {
...
+
+			ApplicationsMenu.gotoPortlet(
+				category = "Product Management",
+				panel = "Commerce",
+				portlet = "Products");
+
+			CPCommerceCatalog.newProductsEntry(
+				productCatalog = "Minium",
+				productName = "Simple T-Shirt",
+				productType = "Simple");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:3649]: Verify if it works with: `CommerceJSONProductsAPI._addCommerceProduct`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 6]
```diff
@@ -3628,6 +3628,73 @@ definition {
...
+				inputValue = "Updated Description",
+				objectLabel = "Commerce Product",
+				thenAction = "Update an Object Entry",
+				whenAction = "On After Update");
+		}
+
+		task ("When the user updates any Commerce Product entry") {
+			CommerceNavigator.gotoPortlet(
+				category = "Product Management",
+				portlet = "Products");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:3674]: Access via URL: `CommerceProducts.openProductsAdmin();`

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
Hi @andreefarias !
All the changes have been made. Can you see it again?

**Review by andreefarias — APPROVED** [review]
Nice @victorhcunha!

LGTM! 🎉

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 7]
```diff
@@ -106,6 +106,55 @@ definition {
 		}
 	}
 
+	@description = "LPS-173537 - Verify deleting a Commerce Product entry triggers an action to add a Commerce Product Group entry"
+	@priority = 5
+	test CanAddCommerceProductGroupEntryAfterDeletingCommerceProductEntry {
+		property portal.acceptance = "true";
+
+		task ("Given a minium site and Commerce Product object") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:114]: Given a Minium site and a Commerce Product

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 8]
```diff
@@ -3623,6 +3672,65 @@ definition {
 		}
 	}
 
+	@description = "LPS-173537 - Verify updating a Commerce Product entry triggers an action to update the Commerce Product entry"
+	@priority = 5
+	test CanUpdateCommerceProductEntryAfterUpdatingCommerceProductEntry {
+		property portal.acceptance = "true";
+
+		task ("Given a minium site and Commerce Product object") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:3680]: Given a Minium site and a Commerce Product

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 9]
```diff
@@ -3623,6 +3672,65 @@ definition {
...
+
+			CommerceNavigator.gotoEntry(entryName = "Simple T-Shirt");
+
+			CommerceEntry.addProductDescription(
+				productFullDescription = "This is a New Full Description",
+				productShortDescription = "This is a New Short Description",
+				publishProduct = "true");
+		}
+
+		task ("Then the action is triggered and the Commerce Product entry changed is updated") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:3723]: Then the action is triggered and the updated Commerce Product entry is updated again

**Review by timpak — CHANGES_REQUESTED** [review]
@victorhcunha just some minor wording updates, you can forward this after you update the strings.

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:foward

**Comment by victorhcunha** [comment]
ci:forward

**Comment by victorhcunha** [comment]
ci:reopen

**Comment by victorhcunha** [comment]
ci:forward

**Pull request closed without merging**